protocols/mdns/src/instance.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use crate::dns::{build_query, build_query_response, build_service_discovery_response};
+use crate::query::MdnsPacket;
+use crate::MdnsConfig;
+use async_io::{Async, Timer};
+use futures::prelude::*;
+use libp2p_core::{address_translation, multiaddr::Protocol, Multiaddr, PeerId};
+use libp2p_swarm::PollParameters;
+use socket2::{Domain, Socket, Type};
+use std::{
+    collections::VecDeque,
+    io, iter,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, UdpSocket},
+    pin::Pin,
+    task::Context,
+    time::{Duration, Instant},
+};
+
+/// A `NetworkBehaviour` for mDNS. Automatically discovers peers on the local network and adds
```

Comment:
well since it's an implementation detail of mdns calling it MdnsInstance is superfluous. I don't think that it's too confusing.

Reply:
I think it would be nice to indicate that this is the state *per interface*. The fact that it is a part of MDns does not need to be in the name, you are right about that.